### ISSUE TITLE
`prefer-regexp-test` catch more cases 

### ISSUE THREAD

**Issue by B2o5T** [sindresorhus/eslint-plugin-unicorn#2071]
the below cases currently not reported

```js
      if (
        uri.match(/unicorn/).length ||
        uri.match(/unicorn/).length > 0 ||
        uri.match(/unicorn/)?.length ||
        uri.match(/unicorn/)?.length > 0
      ) {

      }
```

**Comment by silverwind** [comment]
FWIW there is also https://ota-meshi.github.io/eslint-plugin-regexp/rules/prefer-regexp-test.html but it doesn't detect those either.
